### PR TITLE
Add ICEs from Issue 85103

### PR DESCRIPTION
**Pull request by araruna**
Two examples gathered from [Issue 85103 (in rust-lang/rust)](https://github.com/rust-lang/rust/issues/85103).

### CODE REVIEW

File: ices/85103-2.rs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#![feature(rustc_attrs)]
+
+trait Foo {
+    type Bar;
+}
+
+#[rustc_layout(debug)]
+type Edges<E> = <E as Foo>::Bar;
+
```
Comment (JohnTitor) [inline ices/85103-2.rs:9]:
```suggestion
fn main() {}
```

**Comment by JohnTitor** [comment]
Also, could you drop ices/85103-1.rs and rename ices/85103-2.rs to 85103.rs? it's the minified snippet and we don't have to use both.

**Comment by araruna** [comment]
Yes

> Also, could you drop ices/85103-1.rs and rename ices/85103-2.rs to 85103.rs? it's the minified snippet and we don't have to use both.

Yes. I only kept both of them because I didn't know if the first one would be useful somehow.

**Review by JohnTitor — APPROVED** [review]
Thanks!

**Pull request merged**